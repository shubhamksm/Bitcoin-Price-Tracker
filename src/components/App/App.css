.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  font-family: Roboto;

  background: #303841;
  max-width: 480px;
  margin: auto;
  padding: 60px 30px;
  min-height: 100vh;
}
