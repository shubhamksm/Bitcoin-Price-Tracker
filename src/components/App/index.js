import './App.css';
import React, { useState, useEffect } from 'react';
import Header from '../Header';
import PriceDislay from '../PriceDisplay';
import CurrencySelector from '../CurrencySelector';

function App() {

  const sampleData = {
    "USD": {
        "code": "USD",
        "symbol": "&#36;",
        "rate": "37,138.2632",
        "description": "United States Dollar",
        "rate_float": 37138.2632
    },
    "GBP": {
        "code": "GBP",
        "symbol": "&pound;",
        "rate": "27,341.7093",
        "description": "British Pound Sterling",
        "rate_float": 27341.7093
    },
    "EUR": {
        "code": "EUR",
        "symbol": "&euro;",
        "rate": "30,744.8740",
        "description": "Euro",
        "rate_float": 30744.874
    }
  }

  const sampleCurrencyCodes = ["USD", "GBP", "EUR"];

  const [ selectedCurrency, setCurrency ] = useState({
    code: '',
    rate: ''
  });

  const updateCurrency = code => {
    // TODO : Change with actual Data
    setCurrency({ ...sampleData[code] });
  }

  // Temporary Use Effect to mock stub data
  useEffect(() => {
    setCurrency({ ...sampleData["USD"] });
  }, [])

  return (
    <div className="App">
      <Header />
      <PriceDislay currencyCode={selectedCurrency.code} rate={selectedCurrency.rate} />
      <CurrencySelector 
        updateCurrency={updateCurrency} 
        selectedCurrency={selectedCurrency} 
        currencyCodes={sampleCurrencyCodes} 
      />
    </div>
  );
}

export default App;
