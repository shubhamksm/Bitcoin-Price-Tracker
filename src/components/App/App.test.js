import { render, screen } from '@testing-library/react';
import App from './';

test('Checks App Snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
