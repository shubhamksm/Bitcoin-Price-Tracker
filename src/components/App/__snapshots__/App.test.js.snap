// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks App Snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="Header"
    >
      <svg>
        bitcoin.svg
      </svg>
    </div>
    <div
      class="PriceDisplay"
    >
      <h1>
        Price
      </h1>
      <h1>
        $
         
        37,138.2632
      </h1>
    </div>
    <div
      class="CurrencySelector"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="currency-select currency-select"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="currency-select"
          role="button"
          tabindex="0"
        >
          USD
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value="USD"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
